pygeo/geo_utils/projection.py

```diff
@@ -218,12 +218,14 @@ def projectNode(pt, upVec, p0, v1, v2):
         # This just returns the two points with the minimum absolute
         # distance to the points that have been found.
         fail = -1
-
-        pmin = abs(np.dot(newPoints[:nSol] - pt, upVec))
-        minIndex = np.argsort(pmin)
-
-        return newPoints[minIndex[0]], newPoints[minIndex[1]], fail
-
+        
+        # Account for the direction of the projection
+        pmin = np.dot(newPoints[:nSol] - pt, upVec)
+
+        minIndexDn = np.ma.masked_array(pmin,pmin>=0).argmin()
+        minIndexUp = np.ma.masked_array(pmin,pmin<=0).argmin()
+        
+        return newPoints[minIndexDn], newPoints[minIndexUp], fail 
```

Comment:
For consistency with `nSol == 2`, I would swap the order of the returned points, and merge the check for `fail == -1` in `_generateIntersections` to be the same as `fail == 0`:
https://github.com/mdolab/pygeo/blob/b7f1bfbd01a5638dbd827013906e63498074e65b/pygeo/constraints/DVCon.py#L3309-L3315